packages/drawer/src/mwc-drawer.ts

```diff
@@ -0,0 +1,166 @@
+/**
+@license
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {BaseElement, html, property, observer, query, customElement, Adapter, Foundation, PropertyValues} from '@material/mwc-base/base-element';
+import {classMap} from 'lit-html/directives/classMap.js';
+import MDCModalDrawerFoundation from '@material/drawer/modal/foundation.js';
+import MDCDismissibleDrawerFoundation from '@material/drawer/dismissible/foundation.js';
+import {strings} from '@material/drawer/constants.js';
+//import {MDCModalDrawerFoundation, MDCDismissibleDrawerFoundation, strings, util, createFocusTrap} from '@material/drawer/foundation.js';
+import {style} from './mwc-drawer-css';
+
+export interface DrawerFoundation extends Foundation {
+  open(): void;
+  close(): void;
+}
+
+export declare var DrawerFoundation: {
+  prototype: DrawerFoundation;
+  new(adapter: Adapter): DrawerFoundation;
+}
+
+@customElement('mwc-drawer' as any)
+export class Drawer extends BaseElement {
+
+  @query('.mdc-drawer')
+  mdcRoot!: HTMLElement;
+
+  protected mdcFoundation: MDCDismissibleDrawerFoundation|MDCModalDrawerFoundation|undefined;
```

Comment:
remove `undefined` and make this `mdcFoundation!:`

Reply:
Done